Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp

```diff
@@ -335,9 +337,9 @@ void Recorder::endTransparencyLayer()
 
     appendStateChangeItemIfNecessary();
     recordEndTransparencyLayer();
-    m_stateStack.removeLast();
 
-    m_state.didEndTransparencyLayer(currentState().state.alpha());
+    GraphicsContext::restore();
+    m_stateStack.removeLast();
```

Comment:
Can't we change the order of these two calls to be symmetrical to the calls in `beginTransparencyLayer()`?
```
m_stateStack.removeLast();
GraphicsContext::restore();
```
Also I think we should change `BifurcatedGraphicsContext::endTransparencyLayer()` such that it calls `GraphicsContext::restore()` instead of calling `m_state.didEndTransparencyLayer()`.

Also `GraphicsContextState::didEndTransparencyLayer()` can be removed in this case.

Reply:
Done.